src/main/java/org/isf/medicalinventory/gui/InventoryWardEdit.java

```diff
@@ -105,1002 +104,1603 @@
+                            if (inventory != null) {
+                                MessageDialog.info(null, "angal.inventory.update.success.msg");
+                                resetVariable();
+                                fireInventoryUpdated();
+                                int info = MessageDialog.yesNo(null, "angal.inventory.doyouwanttocontinueediting.msg");
+                                if (info != JOptionPane.YES_OPTION) {
+                                    dispose();
+                                }
+                            } else {
+                                MessageDialog.error(null, "angal.inventory.update.error.msg");
+                                return;
+                            }
+                        } else {
+                            if (!inventoryRowsToDelete.isEmpty()) {
+                                MessageDialog.info(null, "angal.inventory.update.success.msg");
+                                resetVariable();
+                                fireInventoryUpdated();
+                                int info = MessageDialog.yesNo(null, "angal.inventory.doyouwanttocontinueediting.msg");
+                                if (info != JOptionPane.YES_OPTION) {
+                                    dispose();
+                                }
+                            } else {
+                                MessageDialog.info(null, "angal.inventory.inventoryisalreadysaved.msg");
+                                return;
+                            }
+                        }
+                        return;
+                    }
+                    if (!inventory.getInventoryDate().equals(dateInventory)) {
+                        inventory.setInventoryDate(dateInventory);
+                    }
+                    if (!inventory.getUser().equals(user)) {
+                        inventory.setUser(user);
+                    }
+                    if (!lastReference.equals(newReference)) {
+                        inventory.setInventoryReference(newReference);
+                    }
+
+                    MovementType charge = (MovementType) chargeComboBox.getSelectedItem();
+                    inventory.setChargeType(charge != null ? charge.getCode() : null);
+
+                    MovementType discharge = (MovementType) dischargeComboBox.getSelectedItem();
+                    inventory.setDischargeType(discharge != null ? discharge.getCode() : null);
+
+                    Supplier supplier1 = (Supplier) supplierComboBox.getSelectedItem();
+                    inventory.setSupplier(supplier1 != null ? supplier1.getSupId() : null);
+
+                    Ward destination = (Ward) destinationComboBox.getSelectedItem();
+                    inventory.setDestination(destination != null ? destination.getCode() : null);
+
+                    inventory = medicalInventoryManager.updateMedicalInventory(inventory);
```

Comment:
In ward inventory we don't have charge type, discharge type and supplier, so it doesn't make sense ask them to the user